### PR TITLE
[geographiclib] Update to 1.50.1

### LINKED ISSUE #10293: [geographiclib] update to 1.50.1
Library name: GeographicLib

New version number: 1.50.1

Other information that may be useful (release notes, etc...)

https://geographiclib.sourceforge.io/

version 1.50.1 is version update to support the WMM2020 magnetic model expected to be available 2019-12-10.

### PR DESCRIPTION
**Pull request by NancyLi1013**
Update `geographiclib `to the latest version 1.50.1.

Other changes:

- Update the patches
- Update the usage file

Also fix the usage error like this:

```
CMake Error at F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake:329 (_find_package):
1> [CMake]   Could not find a configuration file for package "geographiclib" that is
1> [CMake]   compatible with requested version "".
1> [CMake] 
1> [CMake]   The following configuration files were considered but not accepted:
1> [CMake] 
1> [CMake]     F:/test/vcpkg/installed/x64-windows/share/geographiclib/geographiclib-config.cmake, version: 1.47 (package = GeographicLib, NOT geographiclib)
1> [CMake] 
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeProject8/CMakeLists.txt:5 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
```
Resolve #10293

Note: No feature need to test.

### CODE REVIEW

File: ports/geographiclib/usage [hunk 1]
```diff
@@ -2,7 +2,6 @@ The package geographiclib:x64-linux provides CMake targets:
 
     find_package(GeographicLib CONFIG REQUIRED)
     #dynamic
-    target_link_libraries(main PRIVATE GeographicLib GeographicLib::GeographicLib)
+    target_link_libraries(main PRIVATE GeographicLib GeographicLib_SHARED GeographicLib::GeographicLib GeographicLib::GeographicLib_SHARED)
```
Comment (JackBoosY) [inline ports/geographiclib/usage:5]: Could you please add vcpkg-cmake-wrapper to solve this?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Comment by Mistrical** [comment]
I tried applying this patch to vcpkg:master at commit 7192d3affa4b1d8f88e3e730eb561612c24f7d78, but it failed to apply. See below:
```
$ git apply 11687.patch
11687.patch:46: trailing whitespace.
diff --git a/CMakeLists.txt b/CMakeLists.txt
11687.patch:47: trailing whitespace.
index 4a9ec4e..57bf448 100644
11687.patch:48: trailing whitespace.
--- a/CMakeLists.txt
11687.patch:49: trailing whitespace.
+++ b/CMakeLists.txt
11687.patch:50: trailing whitespace.
@@ -46,6 +46,7 @@ set (LIBVERSION_API 19)
error: patch failed: ports/geographiclib/CONTROL:1
error: ports/geographiclib/CONTROL: patch does not apply
error: patch failed: ports/geographiclib/portfile.cmake:45
error: ports/geographiclib/portfile.cmake: patch does not apply
```

Is the patch invalid or am I doing something wrong?

**Comment by NancyLi1013** [comment]
@Mistrical 
The error is like here:
`11687.patch:46: trailing whitespace.`
You might need to handle `trailing whitespace` issue.

**Comment by Mistrical** [comment]
@NancyLi1013 Is this PR likely to be finished/merged soon?

I'm happy to help with testing it, but I still can't seem to apply the patch to master at commit 70ab27f

**Comment by Septarius** [comment]
@Mistrical Try `git pull upstream pull/11687/head` on the local branch you want to patch. It should work if you have your upstream remote set on your fork. There are conflicts that you can accept incoming on in: `ports/geographiclib/CONTROL` and `ports/geographiclib/portfile.cmake`

**Comment by JackBoosY** [comment]
Please resolve the file conflicts.

**Comment by cffk** [comment]
@NancyLi1013 Hi, I'm the author of geographiclib.  I would be interested in hearing from you how I can simplify the deployment of a vcpkg for geographiclib.  Perhaps a single flag in CMakeLists.txt?  I also have some questions/comments about the patches:

* I see that the tools are not being built.  Perhaps they *should* be, but not installed?  That way the tests could be enabled.

* I've tried to make the version checking in project-config-cmake.cmake.in as robust as possible.  Do you have any suggestions here?

**Comment by NancyLi1013** [comment]
Hi @cffk 
Thanks a lot for coming here. It will be so great with your support and help.

It would be better to add the OPTIONS for these parts:

```
add_subdirectory (tools)
add_subdirectory (man)
add_subdirectory (doc)
add_subdirectory (js)
add_subdirectory (matlab)
add_subdirectory (python/geographiclib)
add_subdirectory (examples)
```
If so, we can disable the codes that we don't need to build via setting the value of options as off instead of making patches.  Or we can add them as features for `geographiclib`. 

I need to investigate why tools are disabled in  current version since it was introduced by some older changes. 

As for the usage, I just added this line:
`set(PROJECT_NAME_LOWER ${PROJECT_NAME})`.

Since the package name seems to be updated `GeographicLib `as `geographiclib`.

We should use `find_package(GeographicLib CONFIG REQUIRED)
` instead of `find_package(geographiclib CONFIG REQUIRED)`.

I will try to rebuild `geographiclib ` and confirm if it can work well after removing patches.

Thanks again for your support.

**Comment by NancyLi1013** [comment]
When I try to use `geographiclib` in release type, it references to a debug file. Since I only copy tools from _bin/_  directory to _tools/geographiclib_ directory. 

```
CMake Error at F:/11687/vcpkg/installed/x64-windows/share/geographiclib/geographiclib-targets.cmake:118 (message):
1> [CMake]   The imported target "CartConvert" references the file
1> [CMake] 
1> [CMake]      "F:/11687/vcpkg/installed/x64-windows/tools/geographiclib/CartConvert_d.exe"
1> [CMake] 
1> [CMake]   but this file does not exist.  Possible reasons include:
1> [CMake] 
1> [CMake]   * The file was deleted, renamed, or moved to another location.
1> [CMake] 
1> [CMake]   * An install or uninstall procedure did not complete successfully.
1> [CMake] 
1> [CMake]   * The installation package was faulty and contained
1> [CMake] 
1> [CMake]      "F:/11687/vcpkg/installed/x64-windows/share/geographiclib/geographiclib-targets.cmake"
1> [CMake] 
1> [CMake]   but not all the files it references.
```

**Comment by cffk** [comment]
Let me try to get my arms around vcpkg and the changes that are being made to the build of GeographicLib.  Then I will be able to offer some intelligent feedback.  (A colleague at work was telling me about vcpkg since a year ago, but I haven't looked at it until now.)  I hope to get back to you within a couple of weeks.

**Comment by strega-nil** [comment]
@cffk if you have any questions about how this stuff works, we're available on discord, slack, and github issues 😄😄😄

**Comment by cffk** [comment]
OK, I've submitted an independent PR #12379.  I've configured this from scratch and expect that the next release of GeographicLib will need a minimal patch to work with vcpkg.  By the way, what's the recommended CMake variable, e.g., `_VCPKG_ROOT` to test for to see if a vcpkg port is being built.

**Comment by NancyLi1013** [comment]
Closed this PR since @cffk has submitted another PR #12379 to update `geographiclib `.

**Pull request closed without merging**